app/src/main/java/pl/pwr/zpi/bcycle/mobile/TripCreationMapActivity.kt

```diff
@@ -0,0 +1,326 @@
+import com.google.firebase.storage.FirebaseStorage
+import com.yarolegovich.lovelydialog.LovelyChoiceDialog
+import com.yarolegovich.lovelydialog.LovelyStandardDialog
+import kotlinx.android.synthetic.main.activity_trip_creation_map.*
+import org.threeten.bp.ZonedDateTime
+import pl.pwr.zpi.bcycle.mobile.api.ApiClient
+import pl.pwr.zpi.bcycle.mobile.models.GroupTrip
+import pl.pwr.zpi.bcycle.mobile.models.GroupTripPoint
+import pl.pwr.zpi.bcycle.mobile.models.UserInfo
+import pl.pwr.zpi.bcycle.mobile.utils.background
+import pl.pwr.zpi.bcycle.mobile.utils.showToast
+import pl.pwr.zpi.bcycle.mobile.utils.showToastError
+
+class TripCreationMapActivity : AppCompatActivity(), OnMapReadyCallback,
+    GoogleMap.OnMarkerClickListener,
+    LocationListener {
+
+    private lateinit var mapFragment: SupportMapFragment
+    private lateinit var lastLocation: Location
+    private var map: GoogleMap? = null
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+    private var myMarkers = mutableListOf<Marker>()
+    private var locationManager: LocationManager? = null
+    private var markerStartPoint: Marker? = null
+    private var markerFinishPoint: Marker? = null
+    // region intent.extra data
+    private lateinit var savedStartDate: ZonedDateTime
+    private lateinit var savedEndDate: ZonedDateTime
+    private lateinit var savedName: String
+    private lateinit var savedDesc: String
+    // endregion intent.extra data
+    private val auth: FirebaseAuth by lazy { FirebaseAuth.getInstance() }
+
+    companion object {
+        private val LOCATION_REQUEST_CODE = 101
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_trip_creation_map)
+        mapFragment = supportFragmentManager
+            .findFragmentById(R.id.myMap) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+        locationManager = getSystemService(Context.LOCATION_SERVICE) as LocationManager
+        setListeners()
+        getSavedData()
+    }
+
+    private fun getSavedData() {
+        val extras = intent.extras;
+        savedDesc = extras.getString(TripCreationActivity.DESCRIPTION_KEY)!!
+        savedName = extras.getString(TripCreationActivity.NAME_KEY)!!
+        savedEndDate = extras.getSerializable(TripCreationActivity.END_DATE_KEY)!! as ZonedDateTime
+        savedStartDate =
+            extras.getSerializable(TripCreationActivity.START_DATE_KEY)!! as ZonedDateTime
+    }
+
+    private fun setListeners() {
+        bt_next.setOnClickListener {
+            LovelyStandardDialog(this)
+                .setTopColorRes(R.color.violet)
+                .setTitle(resources.getString(R.string.prompt_is_trip_done))
+                .setIcon(R.drawable.bike_icon)
+                .setPositiveButton(R.string.yes) {
+                    if(isRequiredDataGiven()){
+                        val photoUrl = FirebaseStorage.getInstance()
```

Comment:
You don’t actually need to pass `UserInfo`, the server doesn’t care about this field (you can just pass `null`).